BackendCommands.php

```diff
@@ -134,43 +118,170 @@ public function updateCode()
+
+        foreach ($exportedFiles as $fileName => $fullPath) {
+            // First check, if the file should be put into the override directory.
+            // otherwise check, if the file should be put into the shared directory.
+            // Finally, if file is new (neither in shared nor in override),
+            // put it into override and inform user of new config.
+            if (isset($overrideFiles[$fileName])) {
+                if (!$this->filesAreEqual($overrideFiles[$fileName], $fullPath)) {
+                    $this->filesystem->copy($fullPath, $overrideFiles[$fileName], true);
+                    $modifiedFiles[$fileName] = $fullPath;
+                }
+            } elseif (isset($sharedFiles[$fileName])) {
+                if (!$this->filesAreEqual($sharedFiles[$fileName], $fullPath)) {
+                    $this->filesystem->copy($fullPath, $sharedFiles[$fileName], true);
+                    $modifiedFiles[$fileName] = $fullPath;
+                }
+            } else {
+                $this->filesystem->copy($fullPath, $overrideFiles[$fileName]);
+                $this->io()->block('New configuration file "'.$fileName.'" was added to override folder. Please check, if that is the correct location.', 'INFO', 'fg=yellow');
+            }
+        }
+
+        // Give information to the user, when we modified a configuration that also exists in local config.
+        // TODO: revisit, when local config folder has been removed.
+        foreach ($modifiedFiles as $fileName => $fullPath) {
+            if (isset($localFiles[$fileName])) {
+                $this->io()->block('Configuration file "'.$fileName.'" was changed and exists in local config folder. Please check, if local config has to be manually modified.', 'INFO', 'fg=yellow');
+            }
         }
 
-        // Move config into shared and site specific folders.
-        // @todo Fix sync-config.sh expecting relative path.
-        $this->process(
-            ['scripts/sync-config.sh', $this->siteConfigSyncDirectory().'/../export'],
-            $this->projectDirectory()
-        );
+        // Remove files from override, that were not exported anymore.
+        foreach ($overrideFiles as $fileName => $fullPath) {
+            if (!isset($exportedFiles[$fileName])) {
+                $this->filesystem->remove($fullPath);
+            }
+        }
+
+        // Remove files from shared, that were not exported anymore.
+        foreach ($sharedFiles as $fileName => $fullPath) {
+            if (!isset($exportedFiles[$fileName])) {
+                $this->filesystem->remove($fullPath);
+                $this->io()->block('Configuration file "'.$fileName.'" was removed from the shared folder. Please check, if overridden config in other sub-sites has to be manually modified or deleted.', 'INFO', 'fg=yellow');
+            }
+        }
+
+        $this->io()->block('Check all config files if they have been moved to the correct location!', 'INFO', 'fg=yellow');
```

Comment:
I think this warning should only be displayed when there were **modifiedFiles**.